requests-for-comments/2022-01-27-future-of-person-model.md

```diff
@@ -0,0 +1,108 @@
+# Request for Comments: Future of person model
+
+- **Created:** 2022-01-27
+- **Owner:** @macobo, @marcushyett-ph 
+
+## Problem: Scaling queries horizontally
+
+In posthog, having insights load fast is key for a great experience. However for large projects tracking >10M active persons per month query performance degrades fast.
+
+This can't be solved by scaling horizontally, because a lot of our queries do aggregation on persons - e.g. filtering by user properties, counting daily active persons and calculating funnels. 
+
+To do these queries we need to JOIN `person` and `person_distinct_id` tables. This in clickhouse is really slow (can cause >30x slower queries) and requires these tables to be replicated on every node, setting an upper bound to scalability. Teams with >10M users often run into memory usage problems as well.
+
+This RFC proposes both a technical and product solution to this problem: Store `person_id` and `person_properties` on the main events table.
+
+However, doing so will require breaking changes to the product. These changes are likely to cause some issues for customers, but these have either a negligible impact (affecting around 0.03% of events) or can be mitigated through documentation and education.
```

Comment:
Do you have links to the analysis done here? Curious to see which users and what kind of events would be affected.

Reply:
Yep - [doc is a little messy](https://docs.google.com/document/d/1RMOoEH-eq8vp2fHqta_JEqgF6Tbd6s7ARhacDpQ5DIo/edit#heading=h.r6voxlxduftz) but theres a data report section on the impact analysis and a section including customer interviews.